src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6567,13 +6567,14 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 ASR::ttype_t* arg_type = nullptr;
                 if( ASR::is_a<ASR::Var_t>(*args[0].m_value) ) {
                     ASR::Var_t* arg_Var = ASR::down_cast<ASR::Var_t>(args[0].m_value);
-                    if( ASR::is_a<ASR::Variable_t>(*arg_Var->m_v) ) {
-                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var->m_v)->m_type;
-                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var->m_v) ) {
+                    ASR::symbol_t *arg_Var2 = ASRUtils::symbol_get_past_external(arg_Var->m_v);
+                    if( ASR::is_a<ASR::Variable_t>(*arg_Var2) ) {
+                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var2)->m_type;
+                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var2) ) {
                         arg_type = ASRUtils::TYPE(ASR::make_Struct_t(al, x.base.base.loc,
-                                        arg_Var->m_v, nullptr, 0));
+                                        arg_Var2, nullptr, 0));
```

Comment:
Actually here you should use `arg_Var->m_v` (i.e., the original external symbol). I am making a new PR with a test.